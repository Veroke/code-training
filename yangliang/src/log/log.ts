type Log = {
    level: string;
    message: any;
};
const logs: Log[] = [];
export function addLog(log:Log) {
    logs.push(log);
}

// 每过一定时间将数据返回至日志服务器
// setTimeout(() => {
//     axios({
//         url: '日志服务器',
//         data: logs
//     })
// },timer)