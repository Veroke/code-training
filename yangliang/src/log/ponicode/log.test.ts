import * as log from "../log"
// @ponicode
describe("log.addLog", () => {
    test("0", () => {
        let result: any = log.addLog({ level: "c466a48309794261b64a4f02cfcc3d64", message: "Error getting key from: %s" })
        expect(result).toMatchSnapshot()
    })

    test("1", () => {
        let result: any = log.addLog({ level: "da7588892", message: "Message recipient is not the grader, the person being graded, or the controller" })
        expect(result).toMatchSnapshot()
    })

    test("2", () => {
        let result: any = log.addLog({ level: "12345", message: "cannot be found." })
        expect(result).toMatchSnapshot()
    })

    test("3", () => {
        let result: any = log.addLog({ level: "12345", message: "New Error " })
        expect(result).toMatchSnapshot()
    })

    test("4", () => {
        let result: any = log.addLog({ level: "bc23a9d531064583ace8f67dad60f6bb", message: "Error:" })
        expect(result).toMatchSnapshot()
    })

    test("5", () => {
        let result: any = log.addLog({ level: "", message: "" })
        expect(result).toMatchSnapshot()
    })
})
