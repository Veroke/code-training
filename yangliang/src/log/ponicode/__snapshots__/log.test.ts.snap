// Vitest Snapshot v1

exports[`log.addLog > 0 1`] = `undefined`;

exports[`log.addLog > 1 1`] = `undefined`;

exports[`log.addLog > 2 1`] = `undefined`;

exports[`log.addLog > 3 1`] = `undefined`;

exports[`log.addLog > 4 1`] = `undefined`;

exports[`log.addLog > 5 1`] = `undefined`;
